src/Symfony/Component/HttpKernel/Config/FileLocator.php

```diff
@@ -65,20 +65,30 @@ public function locate($file, $currentPath = null, $first = true)
             || (\strlen($file) > 3 && ctype_alpha($file[0]) && ':' === $file[1] && ('\\' === $file[2] || '/' === $file[2]))
             || null !== parse_url($file, PHP_URL_SCHEME)
         )) {
+            $triggerDeprecation = false;
+
             // no need to trigger deprecations when the loaded file is given as absolute path
             foreach ($this->paths as $deprecatedPath) {
                 if (\is_array($locations)) {
                     foreach ($locations as $location) {
+                        if (null !== $currentPath && 0 === strpos($location, $currentPath)) {
+                            return $locations;
```

Comment:
Basically, when there are several paths and one of them is not deprecated, we don't know anymore if a deprecated path is going to be used or not, so we skip the deprecation to prevent a false-positive. This means we might miss a deprecation, but that's the best we can do, isn't it?

Reply:
Yes, I am afraid there isn't anything we can do about this. Either we trigger deprecations where we shouldn't or we miss some where they would be appropriate.